ordersettler/ordersettler.go

```diff
@@ -331,19 +331,26 @@ func (r *OrderSettler) SettleBatch(ctx context.Context, batch types.SettlementBa
 			if _, err = q.SetInitiateSettlementTx(ctx, settlementTx); err != nil {
 				return fmt.Errorf("setting initiate settlement tx for settlement from source chain %s with order id %s: %w", settlement.SourceChainID, settlement.OrderID, err)
 			}
+		}
 
-			submittedTx := db.InsertSubmittedTxParams{
-				OrderSettlementID: sql.NullInt64{Int64: settlement.ID, Valid: true},
-				ChainID:           settlement.DestinationChainID,
-				TxHash:            txHash,
-				RawTx:             rawTx,
-				TxType:            dbtypes.TxTypeSettlement,
-				TxStatus:          dbtypes.TxStatusPending,
-			}
-			if _, err = q.InsertSubmittedTx(ctx, submittedTx); err != nil {
-				return fmt.Errorf("inserting raw tx for settlement with hash %s: %w", txHash, err)
-			}
+		// we do not insert a submitted tx for each settlement, since many
+		// settlements are settled by a single tx (batch settlements)
+
+		submittedTx := db.InsertSubmittedTxParams{
```

Comment:
i added settlement id as a 3rd field in the  submittedTx table constraint to fix this issue earlier, but this makes more sense since its one tx per batch. thanks for fixing this

https://github.com/skip-mev/skip-go-fast-solver/blob/3e5ef6dce616585e9370a09c3303988c00ae1f2b/db/migrations/000004_add_submitted_tx_table.up.sql#L23